servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/ContentCodings.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.encoding.netty;
+
+import io.servicetalk.encoding.api.ContentCodec;
+import io.servicetalk.encoding.netty.DefaultContentCodecBuilder.ZipContentCodecBuilder;
+
+/**
+ * Common available encoding implementations.
+ */
+public final class ContentCodings {
+
+    private static final ContentCodec DEFAULT_GZIP = gzip().build();
+
+    private static final ContentCodec DEFAULT_DEFLATE = deflate().build();
+
+    private ContentCodings() {
+    }
+
+    /**
+     * Returns the default, always supported 'identity' {@link ContentCodec}.
+     * @return the default, always supported 'identity' {@link ContentCodec}
+     */
+    public static ContentCodec identity() {
+        return io.servicetalk.encoding.api.ContentCodings.identity();
```

Comment:
If io.servicetalk.encoding.api.ContentCodings is to be deprecated perhaps the indentity constant there should reference this one rather than the other way around. That way when it is removed no changes are needed here.

Reply:
We typically don't have `api` depend upon `netty` modules (to avoid leaking Netty into API), but is the intention to move the `IdentityContentCodec` into the `netty` package so we can completely remove `io.servicetalk.encoding.api.ContentCodings`? If so lets mark this class as deprecated too.

Reply:
The intention was to keep this one around because it's not bound to an implementation (ie. Netty) and its used in a few places as ref-check to skip coding headers and/or skip encoding/decoding see. `io.servicetalk.encoding.api.internal.HeaderUtils`
So the goal was to only deprecate the actual implementation of ` io.servicetalk.encoding.api.ContentCodings` but not `identity`.

Reply:
Instead of exposing a new `API` and specialized code (e.g. single element in collection, etc.) can we just use `null` for this?